low-level/files/opt/seagate/sspl/setup/sspl_post_install.py

```diff
@@ -0,0 +1,156 @@
+        # on prod. For node replacement scenario consul will not be running on the new node. But,
+        # there will be two instance of consul running on healthy node. When new node is configured
+        # consul will be brought back on it. We are using VIP to connect to consul. So, if consul
+        # is not running on new node, we dont need to error out. So, need to skip this step for 
+        # node replacement case
+        # Onward LDR_R2, consul will be abstracted out and it won't exit as hard dependeny of SSPL
+        if PRODUCT_NAME == "LDR_R1":
+            # setup consul if not running already
+            if not os.path.exists(REPLACEMENT_NODE_ENV_VAR_FILE):
+                SSPLSetup._call_script(f"{self._script_dir}/sspl_setup_consul", ["-e", f"{ENVIRONMENT}"])
+
+        # Install packages which are not available in YUM repo, from PIP
+        pip_cmd = f"python3 -m pip install -r {SSPL_BASE_DIR}/low-level/requirements.txt"
+        subprocess.call(pip_cmd.split(" "), shell=False)
+
+        # NOTE: By default the sspl default conf file will not be copied.
+        # The provisioner is supposed to copy the appropriate conf file based
+        # on product/env and start SSPL with it.
+        # TODO: Disable this default copy once the provisioners are ready.
+        if not os.path.exists(file_store_config_path):
+            shutil.copyfile(f"{SSPL_BASE_DIR}/conf/sspl.conf.{PRODUCT}", file_store_config_path)
+        
+        # For Dev environment
+        if ENVIRONMENT == "DEV":
+            shutil.copyfile(f"{SSPL_BASE_DIR}/conf/sspl.conf.{PRODUCT}", file_store_config_path)
+
+        # Copy rsyslog configuration
+        if not os.path.exists(self.RSYSLOG_CONF):
+            shutil.copyfile(f"{SSPL_BASE_DIR}/low-level/files/{self.RSYSLOG_CONF}", self.RSYSLOG_CONF)
+
+        if not os.path.exists(self.RSYSLOG_SSPL_CONF):
+            shutil.copyfile(f"{SSPL_BASE_DIR}/low-level/files/{self.RSYSLOG_SSPL_CONF}", self.RSYSLOG_SSPL_CONF)
+
+        # Create soft link for SINGLE product name service to existing LDR_R1, LDR_R2 service
+        # Instead of keeping separate service file for SINGLE product with same content.
+        currentProduct = f"{SSPL_BASE_DIR}/conf/sspl-ll.service.{PRODUCT}"
+        print(f"currentProduct name {currentProduct}")
+        if (PRODUCT == "SINGLE" and not os.path.exists(currentProduct)) or (PRODUCT == "DUAL" and not os.path.exists(currentProduct)):
+            os.symlink(f"{SSPL_BASE_DIR}/conf/sspl-ll.service.{PRODUCT_NAME}", currentProduct)
+
+        if PRODUCT == "CLUSTER" and not os.path.exists(currentProduct):
+            os.symlink(f"{SSPL_BASE_DIR}/conf/sspl-ll.service.LDR_R2", currentProduct)
+        
+        # Copy sspl-ll.service file and enable service
+        shutil.copyfile(currentProduct, "/etc/systemd/system/sspl-ll.service")
+        SSPLSetup._send_command("systemctl enable sspl-ll.service")
```

Comment:
use dbus interface to control services. Not sure all can be satisfied through that, like daemon reload, so where ever dbus api not available, use sub-process there alone.

Reply:
Done